plugins/inputs/windows_event_log/windows_event_log.go

```diff
@@ -72,8 +73,18 @@ func (s *Plugin) FindLogSrc() []logs.LogSrc {
  * We can do any initialization in this method.
  */
 func (s *Plugin) Start(acc telegraf.Accumulator) error {
+	err := os.MkdirAll(s.FileStateFolder, 0755)
+	if err != nil {
+		s.Log.Errorf("failed to create state file directory %s: %v",
+			s.FileStateFolder, err)
+		return err
+	}
 	for _, eventConfig := range s.Events {
-		stateFilePath := logscommon.WindowsEventLogPrefix + escapeFilePath(eventConfig.LogGroupName)
+		// Assume no 2 EventConfigs have the same combination of:
```

Comment:
Is this a safe assumption to make? If there is a contract that ensures this always holds true, is it documented?